libs/@local/hash-graph-types/graph_types/data_type.py

```diff
@@ -96,57 +96,27 @@ async def create_data_type(
         *,
         actor_id: UUID,
         graph: "GraphAPIProtocol",
-    ) -> Annotated[Any, "DataTypeAnnotation"]:
+    ) -> type[DataTypeBase]:
         """Create an annotated type from this schema."""
         # Custom data types will require an actor ID and the graph to be passed in
         _actor_id = actor_id
         _graph = graph
 
         const = self.model_extra.get("const") if self.model_extra else None
 
-        # TODO: Use `Field` instead when multiple fields are supported
-        #   https://github.com/pydantic/pydantic/issues/6349
-        #   https://github.com/pydantic/pydantic/issues/6353
-        class DataTypeAnnotation:
-            @classmethod
-            def __get_pydantic_core_schema__(
-                cls,
-                source_type: Any,  # noqa: ANN401
-                handler: GetCoreSchemaHandler,
-            ) -> CoreSchema:
-                schema = handler(source_type)
-                if const is not None:
-                    return core_schema.no_info_after_validator_function(
-                        cls.validate_const,
-                        schema,
-                    )
-                return schema
-
-            @classmethod
-            def __get_pydantic_json_schema__(
-                cls,
-                schema: CoreSchema,
-                handler: GetJsonSchemaHandler,
-            ) -> JsonSchemaValue:
-                json_schema = handler(schema)
-                json_schema.update(
-                    **{
-                        "$id": self.identifier,
-                        "$schema": self.schema_url,
-                        "title": self.title,
-                        "description": self.description,
-                        "kind": self.kind,
-                    },
-                )
-                if const is not None:
-                    json_schema.update(const=const)
-
-                return json_schema
-
-            @classmethod
-            def validate_const(cls, v: DataType) -> None:
-                if v != const:
-                    msg = f"Value must be {const}"
-                    raise ValueError(msg)
-
-        return Annotated[self._type(), DataTypeAnnotation]
+        type_ = self._type()
+        if const:
```

Comment:
`const` could be `False`
```suggestion
        if const is not None:
```

Reply:
or even better check if `const` is in `model_extra`

Reply:
or `[]` or an empty string or anything else. Yea major oversight. Thanks for the flag.

Reply:
> or even better check if `const` is in `model_extra`

I'll do that, I did that before but mypy yelled at me because `model_extra` can be `None`

Reply:
```python
if "const" in (self.model_extra or {})
```
😂

Reply:
I have that exact line in my working copy right now xD

Reply:
fixed in [4523167](https://github.com/hashintel/hash/pull/2776/commits/45231679687f7011718077219d96d8dd473f13f7)